internal/pkg/aws/codepipeline/codepipeline_test.go

```diff
@@ -505,3 +505,162 @@ func TestCodePipeline_GetPipelineState(t *testing.T) {
 		})
 	}
 }
+
+func TestCodePipeline_ListPipelineExecution(t *testing.T) {
+	mockPipelineName := "pipeline-dinder-badgoose-repo"
+	mockErr := errors.New("some error")
+	mockOutput := &codepipeline.ListPipelineExecutionsOutput{
+		PipelineExecutionSummaries: []*codepipeline.PipelineExecutionSummary{
+			{
+				PipelineExecutionId: aws.String("12345678-fake-exec-utio-nid987654321"),
+			},
+		},
+	}
+
+	tests := map[string]struct {
+		callMocks     func(m codepipelineMocks)
+		expectedOut   *string
+		expectedError error
+	}{
+		"happy path": {
```

Comment:
What happens if this returns a zero-length list? ie there are no pipeline executions.

Reply:
good catch. thanks!